sidebars: Place new heading typography, icons, and partial colors.

karlstolley opened the pull request:
This PR implements new heading typography and expand/collapse icons from @terpimost's [sidebar redesign](https://terpimost.github.io/zulip-sidebar/), as well as new colors on the headings and subheadings.

[CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/proposed.20left-sidebar.20spacing.20and.20alignment/near/1922052)

There are a few things not yet implemented from Vlad's design, all of which are effects-based:

* Hover styles and triangle-rotation effects are not yet implemented on the buddy list subheaders; those will require reworking how buddy lists are added and removed from the DOM, even on clicking expand/collapse
* Drop shadows on the buddy list subheader are not yet implemented; that represents a bit too much churn in an already quite large PR

**Screenshots and screen captures** (updated 2024-09-09):

| Before | After |
| --- | --- |
| ![headings-20240909-light-before](https://github.com/user-attachments/assets/f4b7bb51-e9f4-453b-8f5f-65ea4c768d0e) | ![headings-20240909-light-after](https://github.com/user-attachments/assets/c8e68709-fc63-4e1f-96eb-391be8c8f043) |
| ![headings-20240909-dark-before](https://github.com/user-attachments/assets/15ab8bd6-80d2-4c51-9677-555a6eb8ba89) | ![headings-20240909-dark-after](https://github.com/user-attachments/assets/1a4030c1-f6d8-4d9c-a736-753e334043d9) |

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Cool! Playing with the PR, something feels off about the "All topics" styling. I think the character spacing is too wide? And maybe the font is too large? I'm less sure about the latter part.

alya commented:
I worry a bit that changing the "All topics" font without changing the color might make it harder to tell that it's a button -- it'll blend in with the topic names more. Maybe it'll be OK.

timabbott commented:
The other thing I noticed is that the "in this conversation" type labels in the right sidebar seemed at the edge of insufficient contrast.

alya commented:
For reference, here's a screenshot of the DMs section. I think it's fine, but useful to look at, as it's not quite the same as channels/topics.

![Screenshot 2024-08-27 at 13 06 17@2x](https://github.com/user-attachments/assets/5bb64eda-4b2c-40ab-8721-9d857d61e74a)

alya commented:
The tooltip for the "In this channel" section is incorrect -- I think it's a regression in this PR.

![Screenshot 2024-08-27 at 13 08 02@2x](https://github.com/user-attachments/assets/0309dea7-f5a4-4484-895b-6549ec4d507f)

alya commented:
That's all I've got!

karlstolley commented:
> For reference, here's a screenshot of the DMs section. I think it's fine, but useful to look at, as it's not quite the same as channels/topics.

@alya are you referring to the letter spacing? the alignment is different in Vlad's design for MORE CONVERSATIONS (although I see now that Vlad uses the copy ALL DIRECT MESSAGES, which better resonates with the ALL TOPICS copy)

karlstolley commented:
Okay. I've addressed all of the feedback above with the following changes and fixes:

- [x] The **In this channel** tooltip displays as expected
- [x] I've recalculated the metrics for `letter-spacing` on ALL TOPICS and friends; Vlad's values were based on a 15px em, so I've carried that precise value forward for a 16px em. The result is a smidge tighter letter spacing.
- [x] I've placed the color for ALL TOPICS and friends; hover styles remain the same, but will be addressed in their redesigned form in a near-future PR 
- [x] I've changed the "more conversations" label to ALL DIRECT MESSAGES, both implementing the wording of Vlad's mock as well as paralleling ALL TOPICS.
- [x] I agree that the opacity on the right sidebar subheaders is too light; I've bumped those up from 0.5, as used in Vlad's mock, to 0.65.

alya commented:
> I've changed the "more conversations" label to ALL DIRECT MESSAGES, both implementing the wording of Vlad's mock as well as paralleling ALL TOPICS.

Let's undo this part. See #30014.

alya commented:
We should also try going back to "show all topics" as the text; cf. https://github.com/zulip/zulip/pull/30033#issuecomment-2116107718.

alya commented:
In general, we should not use Vlad's designs as an authoritative reference for what strings we want.

karlstolley commented:
Got it. Pushed a fresh set of commits with all of the mistaken wording changes reverted.

alya commented:
Testing it again, let's at least make the "ALL TOPICS" text color stay the same when hovered, rather than turning white in dark theme. "MORE CONVERSATIONS" stays purple, which looks OK.

![Screenshot 2024-09-09 at 23 11 01@2x](https://github.com/user-attachments/assets/4b821c9d-f1f5-49bd-bb64-d17e255d0ff1)
![Screenshot 2024-09-09 at 23 11 07@2x](https://github.com/user-attachments/assets/bf9fb67c-6eee-49e0-bc97-ca27bf20f2a0)

alya commented:
For some reason all the views links look blue in when I test (light and dark theme).

![Screenshot 2024-09-09 at 23 06 03@2x](https://github.com/user-attachments/assets/19dc6864-a67e-4fcd-bfa5-9f15f96671de)

karlstolley commented:
Awesome, thank you for catching those, @alya. I've pushed some fresh commits that should fix both the hover color on the SHOW ALL TOPICS heading as well as the incorrect blue colors on the navigation items.

karlstolley commented:
Pushed an additional commit to share the action-heading styles with View All Subscribers/Users links:

![Screen Shot 2024-09-10 at 14 41 54](https://github.com/user-attachments/assets/a9f3e5d4-e1a0-410f-8d32-5bf8f4355a9c)

timabbott commented:
Looks great, marked to merge once CI passes, thanks @karlstolley!
